main_test.go

```diff
@@ -20,9 +24,69 @@ func TestMain(m *testing.M) {
 	os.Exit(m.Run())
 }
 
+//////
+// utility functions used in tests
+
+// New consul test process
 func NewTestServer() (*testutil.TestServer, error) {
 	return testutil.NewTestServerConfig(func(c *testutil.TestServerConfig) {
 		c.Stdout = STDOUT
 		c.Stderr = STDERR
 	})
 }
+
+// Show diff between structs
+func structDiff(act, exp interface{}) string {
+	var b strings.Builder
+	b.WriteString("\n")
+	va := valueOf(act)
+	ve := valueOf(exp)
+	ct := ve.Type()
+
+	for i := 0; i < ve.NumField(); i++ {
+		if !ve.Field(i).CanInterface() {
+			continue
+		}
+		fe := ve.Field(i).Interface()
+		fa := va.Field(i).Interface()
+		if !reflect.DeepEqual(fe, fa) {
+			fmt.Fprintf(&b, "%s: ", ct.Field(i).Name)
+			fmt.Fprintf(&b, "got: %#v", fa)
+			fmt.Fprintf(&b, ", want: %#v", fe)
+		}
+	}
+
+	return b.String()
+}
+
+// Compare 2 api.HealthChecks objects for important differences
+func compareHealthCheck(c1, c2 *api.HealthCheck) error {
+	fields := []string{"Node", "CheckID", "Name", "Status", "Output"}
```

Comment:
Could we blacklist only CreateIndex/ModifyIndex here instead of whitelisting fields? Then it'd be impossible to forget to add a field here when there's something new in the api.HealthCheck struct.

Reply:
I was using the fields from the `expected` test object as the baseline, but I'll try that and let you know if it works.

Reply:
Worked fine except I also added ServiceTags to the skipped fields. It is a slice, so DeepEqual doesn't work as well, and it is always empty in tests.

Reply:
Just pushed the change. Uses skip/black list instead of the whitelist.